pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -210,14 +214,58 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 		}
 		return r.updateStatefulSet(instance, foundSts, sts)
 	}
-	return nil
 }
 
 func (r *ReconcileZookeeperCluster) updateStatefulSet(instance *zookeeperv1beta1.ZookeeperCluster, foundSts *appsv1.StatefulSet, sts *appsv1.StatefulSet) (err error) {
```

Comment:
`updateStatefulSet` is handling both scale up/down as well as upgrade. Could we modularize the code to distinguish the handling of both these scenarios?

Reply:
I have added function **upgradeStatefulSet()**  which has the upgrade logic now.